packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -29,13 +29,25 @@ export const DataTableComponent = <T extends RecordWithId>({
   noDataMessage,
   ExpandContent,
   dense = false,
+  isDisabled = false,
 }: TableProps<T>): JSX.Element => {
   const t = useTranslation('common');
-  const { setActiveRows } = useTableStore();
+  const { setActiveRows, setDisabledRows, rowState } = useTableStore();
+
+  console.log(rowState);
+
   useEffect(() => {
-    if (data.length) setActiveRows(data.map(({ id }) => id as string));
+    if (data.length) setActiveRows(data.map(({ id }) => id));
   }, [data]);
 
+  useEffect(() => {
+    console.log(
+      'setting disabled',
+      data.map(({ id }) => id)
+    );
+    if (isDisabled) setDisabledRows(data.map(({ id }) => id));
+  }, [isDisabled, data]);
+
```

Comment:
Not sure if this is the best idea, but.. it was easy to implement!

Essentially passing `isDisabled` into the `DataTable` disables all rows